plutus-core-spec/figures/PlutusCoreReduction.tex

```diff
@@ -60,6 +61,7 @@
                        %&   &     & \inLamLeftFrame{x}{M}                   & \textrm{$\lambda$}\\
                        &   &     & \inAppLeftFrame{M}                      & \textrm{left app}\\
                        &   &     & \inAppRightFrame{V}                     & \textrm{right app}\\
+                       &   &     & \inBuiltin{bn}{A*}{V^*}{\_}{M^*}        & \textrm{builtin}\\
```

Comment:
Since here we have n-ary type instantiations and applications, would it make sense to have a hole for each of the term arguments? I.e. instead of one `_`, we could have an entire `_^*` and fill them all at the same time. I think it would simplify both the spec and the implementation. Or does this not make sense?

Reply:
It doesn't make sense because this is the mechanism for  imposing reduction order, which is not parallel.

Reply:
Does it? It ought to be something along the lines of

```
normalize env (Builtin bn _ ms) = normalizeBuiltin bn  <$> mapM (normalize env) ms
```

Reply:
That's what I thought of when I asked about "n-ary holes". But the implementations of the CK and CEK machines seem to become more complicated. I'm not entirely sure.